File: 3651-transformed-array/3651-transformed-array.cs
public class Solution {
    public int[] ConstructTransformedArray(int[] nums) {
        int n = nums.Length;
        int[] res = new int[n];
        int j = 0;
        for (int i = 0; i < n; ++i) {
            int idx = (i + nums[i] % n + n) % n;
            res[j++] = nums[idx];
        }
        return res;
    }
}

File: 3651-transformed-array/3651-transformed-array.js
/**
 * @param {number[]} nums
 * @return {number[]}
 */
var constructTransformedArray = function(nums) {
    const res = []
    let n = nums.length;
    for (let i = 0; i < n; i++) {
        let idx = (i + nums[i] % n + n) % n;
        res.push (nums[idx]);
    }

    return res;
};